source/time_tables/departure_boards.cpp

```diff
@@ -113,11 +127,29 @@ render(PbCreator& pb_creator,
                 }
             }
         }
+
+        // add first datetime
+        const auto& first_it = map_route_first_point.find(id_vec.first);
```

Comment:
I see duplicated code here !

Reply:
Yes it is, because we have 2 maps, one for `first` and one for `last`, and the key depends is a particular route point. Thus, we have same code for both, but not on the same map.
We can find a way to reduce code, `map_route_first_point` and `map_route_last_point` can become a map with a pair like : 

```
map<pair<first,last>>
```
I'll change that